contracts/StablecoinConverter.sol

```diff
@@ -573,8 +581,8 @@ contract StablecoinConverter is EpochTokenLocker {
       */
     function checkAndOverrideObjectiveValue(uint256 newObjectiveValue) private {
         require(
-            newObjectiveValue > getCurrentObjectiveValue(),
-            "Solution must have a higher objective value than current solution"
+            newObjectiveValue.mul(IMPROVEMENT_DENOMINATOR) > getCurrentObjectiveValue().mul(IMPROVEMENT_DENOMINATOR + 1),
+            "New objective doesn't sufficiently improve current solution"
```

Comment:
Is there a reason to duplicate this code? Can we put it in a helper method? 

Alternatively, we could also verify that the claimed objective value equals the computed value (at the end). But maybe that's too restrictive?

Reply:
This is a great point you have brought up. In fact we can solve both problems here with one fix! At first I would have said "I don't think it's necessary to verify that the claimed value equals the computed and that we need it to get full branch coverage", but this was actually just a silly mistake when first introducing the claimedValue. The problem here is actually the robustness of the existing helper method `checkAndOverrideObjectiveValue` which does the inequality check and updates the storage. When first introducing the check for claimed objective, there was no way around checking again if we wanted to make use of the helper method! Furthermore, this smell made its way into our tests in places where we were forced to claim a knowingly false objective value in order to get past the first check and fail elsewhere.

I think there is an elegant solution here, and I will get started on the fix right away... Warning! It will require moving the token conservation check up by two lines.

Reply:
BAM! https://github.com/gnosis/dex-contracts/pull/344